storey/flow.py

```diff
@@ -1634,33 +1798,20 @@ async def _do(self, event):
             futures = []
             runnables_encountered = set()
             for runnable in runnables:
-                if isinstance(runnable, str):
-                    runnable = self._runnable_by_name[runnable]
-                if id(runnable) in runnables_encountered:
-                    raise ValueError(f"select_runnables() returned more than one outlet named '{runnable.name}'")
-                input = event.body if runnable.execution_mechanism == "multiprocessing" else copy.deepcopy(event.body)
-                runnables_encountered.add(id(runnable))
-                if runnable.execution_mechanism == "asyncio":
-                    future = asyncio.get_running_loop().create_task(runnable._async_run(input, event.path))
-                elif runnable.execution_mechanism == "naive":
-                    future = asyncio.get_running_loop().create_future()
-                    future.set_result(runnable._run(input, event.path))
-                elif runnable.execution_mechanism == "dedicated_process":
-                    executor = self._process_executor_by_runnable_name[runnable.name]
-                    future = asyncio.get_running_loop().run_in_executor(
-                        executor,
-                        _static_run,
-                        input,
-                        event.path,
+                runnable: ParallelExecutionRunnable = (
+                    runnable
+                    if isinstance(runnable, ParallelExecutionRunnable)
+                    else self.runnable_executor._runnable_by_name[runnable]
+                )
+                if runnable.execution_mechanism == ParallelExecutionMechanisms.shared_executor:
+                    future = self.context.executor.run_executor(
+                        runnable=runnable.name, runnables_encountered=runnables_encountered, event=event
```

Comment:
Only the executor differs:
```suggestion
                executor = (
                    self.context.executor
                    if runnable.execution_mechanism == ParallelExecutionMechanisms.shared_executor
                    else self.runnable_executor
                )
                future = executor.run_executor(
                    runnable=runnable, runnables_encountered=runnables_encountered, event=event
                )
```